File: meeting/reservation.py
class Reservation:
    def __init__(self, manager, name, phone_number, email, time_slot, purpose):
        self.manager = manager
        self.name = name
        self.phone_number = phone_number
        self.email = email
        self.time_slot = time_slot
        self.purpose = purpose

class ReservationManager:
    def __init__(self, id, name, phone_number, email, permission):
        self.id = id
        self.name = name
        self.phone_number = phone_number
        self.email = email
        self.permission = permission

        # 予約情報のリスト
        self.reservations = []

    def add_reservation(self, reservation):
        """
        予約を追加する
        """
        self.reservations.append(reservation)

    def remove_reservation(self, reservation):
        """
        予約を削除する
        """
        self.reservations.remove(reservation)

    def get_reservations(self):
        """
        登録されているすべての予約を取得する
        """
        return self.reservations

File: meeting_java/app/src/main/java/org/autotaker/gpt_gen/meeting/reservations/Reservation.java
package org.autotaker.gpt_gen.meeting.reservations;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import java.time.Instant;
import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Data
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reservationId;
    private String name;
    private String email;
    private String phoneNumber;
    private LocalDateTime timeSlot;
    private String purpose;
    private Long managerId;
    @CreatedDate
    private Instant createdAt;
    @LastModifiedDate
    private Instant updatedAt;

    @Builder
    private Reservation(String name, String email, String phoneNumber, LocalDateTime timeSlot, String purpose,
            ReservationManager manager) {
        this.name = name;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.timeSlot = timeSlot;
        this.purpose = purpose;
        this.managerId = manager.getManagerId();
    }
}
